services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -233,20 +285,19 @@ async def delete_files_in_project_node(
     ) -> None:
         # NOTE: the / at the end of the Prefix is VERY important,
         # makes the listing several order of magnitudes faster
-        response = await self.client.list_objects_v2(
-            Bucket=bucket,
-            Prefix=f"{project_id}/{node_id}/" if node_id else f"{project_id}/",
+        s3_objects: list[ObjectTypeDef] = await _list_objects_v2_all_items(
```

Comment:
ok I have a few questions here.
1. I see you make now list_files use your new *non-paginated* function. would it not make more sense to make s3_client.py provide the pagination parameters and the client code take the assumption of the max 1000 fileS?
2. how do you ensure deleting a path will now work? since it is caped at 1000 files and not paginating, the folders will not be entirely deleted right? -> delete is broken.
3. why do you have 2 delete functions now? I don't get that. the client code knows it needs to delete inside the path. I would suggest using a recursive parameter or similar instead. we can also discuss tomorrow.